cpp/include/cudf/io/types.hpp

```diff
@@ -166,6 +168,21 @@ struct host_buffer {
   host_buffer(const char* data, size_t size) : data(data), size(size) {}
 };
 
+/**
+ * @brief Returns `true` if the type is byte-like, meaning it is reasonable to pass as a pointer to
+ * bytes.
+ *
+ * @tparam T The representation type
+ * @return `true` if the type is considered a byte-like type
+ */
+template <typename T>
+constexpr inline auto is_byte_like_type()
+{
+  return cuda::std::is_same_v<T, int8_t> || cuda::std::is_same_v<T, char> ||
+         cuda::std::is_same_v<T, uint8_t> || cuda::std::is_same_v<T, unsigned char> ||
+         cuda::std::is_same_v<T, std::byte>;
```

Comment:
We can make `is_byte_like_type` more robust :)
```suggestion
{
  using non_cv_T = std::remove_cv_t<T>;
  return std::is_same_v<non_cv_T , int8_t> || std::is_same_v<non_cv_T , char> ||
         std::is_same_v<non_cv_T , uint8_t> || std::is_same_v<non_cv_T , unsigned char> ||
         std::is_same_v<non_cv_T , std::byte>;
```

Looks like std library includes `is_same_v` in C++17 onward, we don't need CUDA version.